packages/site/src/pages/BountyDetail/ChildBountyDetail/index.js

```diff
@@ -20,36 +23,94 @@ import DetailGoBack from "../../components/DetailGoBack";
 import { useChainRoute } from "../../../utils/hooks";
 import ChildInformationTable from "./ChildInformationTable";
 import { processTimeline } from "../index";
+import ClaimButton from "./ClaimButton";
+import { newPendingToast, newSuccessToast, newToastId, removeToast } from "../../../store/reducers/toastSlice";
+import { sleep } from "../../../utils";
+import useApi from "../../../hooks/useApi";
 
 const ChildBountyDetail = () => {
   useChainRoute();
 
   const { bountyIndex } = useParams();
   const dispatch = useDispatch();
   const [timelineData, setTimelineData] = useState([]);
+  const api = useApi();
 
   const symbol = useSelector(chainSymbolSelector);
   const chain = useSelector(chainSelector);
 
   useEffect(() => {
     dispatch(fetchChildBountyDetail(chain, bountyIndex));
     return () => {
-      dispatch(fetchChildBountyDetail({}));
+      dispatch(setChildBountyDetail({}));
     };
   }, [dispatch, chain, bountyIndex]);
 
   const loadingBountyDetail = useSelector(loadingBountyDetailSelector);
   const bountyDetail = useSelector(childBountyDetailSelector);
   const scanHeight = useSelector(scanHeightSelector);
 
+  const refScanHeight = useRef();
+  useEffect(() => {
+    refScanHeight.current = scanHeight;
+  }, [scanHeight]);
+
   useEffect(() => {
     setTimelineData(processTimeline(bountyDetail, scanHeight, symbol));
   }, [bountyDetail, scanHeight, symbol]);
 
+  const waitScanAndUpdate = useCallback(
+    async (blockHash) => {
+      dispatch(newSuccessToast("Rewards claimed", 1000));
+
+      if (!api) {
+        return;
+      }
+
+      const toastId = newToastId();
+      setTimeout(() => {
+        dispatch(newPendingToast(toastId, "Waiting to sync on-chain data..."));
+      }, 1000);
+
+      const block = await api.rpc.chain.getBlock(blockHash);
+      const targetHeight = block.block.header.number.toNumber();
+
+      try {
+        let times = 6;
+        while (true) {
+          times--;
+          if (times === 0) {
+            return;
+          }
+          await sleep(10000);
+          if (refScanHeight.current >= targetHeight) {
+            break;
```

Comment:
We still need to query detail if we find the scan height is greater than target height before the break clause.